plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/impl/BaseFlightProducer.java

```diff
@@ -0,0 +1,266 @@
+                        .toRuntimeException();
+                    throw unexpectedEx;
+            }
+        };
+    }
+
+    private void handleCancellation(StreamProducer.BatchedJob<VectorSchemaRoot> batchedJob, ServerStreamListener listener) {
+        try {
+            batchedJob.onCancel();
+            FlightRuntimeException ex = CallStatus.CANCELLED.withDescription("Stream cancelled before processing").toRuntimeException();
+            throw ex;
+        } catch (Exception e) {
+            logger.error("Unexpected error during cancellation", e);
+            FlightRuntimeException fre = CallStatus.INTERNAL.withCause(e)
+                .withDescription("Error during cancellation: " + e.getMessage())
+                .toRuntimeException();
+            throw fre;
+        }
+    }
+
+    private FlightInfo getLocalFlightInfo(StreamTicket streamTicket, FlightDescriptor descriptor) {
+        FlightStreamManager.StreamProducerHolder producerHolder = streamManager.getStreamProducer(streamTicket).orElseThrow(() -> {
+            logger.debug("FlightInfo not found for ticket: {}", streamTicket);
+            return CallStatus.NOT_FOUND.withDescription("FlightInfo not found").toRuntimeException();
+        });
+
+        Location location = flightClientManager.getFlightClientLocation(streamTicket.getNodeId()).orElseThrow(() -> {
+            logger.debug("Failed to determine location for node: {}", streamTicket.getNodeId());
+            return CallStatus.UNAVAILABLE.withDescription("Internal error determining location").toRuntimeException();
+        });
+
+        try {
+            Ticket ticket = new Ticket(descriptor.getCommand());
+            var schema = producerHolder.getRoot().getSchema();
+            FlightEndpoint endpoint = new FlightEndpoint(ticket, location);
+            return FlightInfo.builder(schema, descriptor, Collections.singletonList(endpoint))
+                .setRecords(producerHolder.producer().estimatedRowCount())
+                .build();
+        } catch (Exception e) {
+            logger.error("Failed to build FlightInfo", e);
+            throw CallStatus.INTERNAL.withCause(e).withDescription("Error creating FlightInfo: " + e.getMessage()).toRuntimeException();
+        }
+    }
+
+    private FlightInfo getRemoteFlightInfo(StreamTicket streamTicket, FlightDescriptor descriptor) {
+        FlightClient remoteClient = flightClientManager.getFlightClient(streamTicket.getNodeId()).orElseThrow(() -> {
+            logger.warn("No remote client available for node: {}", streamTicket.getNodeId());
+            return CallStatus.UNAVAILABLE.withDescription("Client doesn't support Stream").toRuntimeException();
+        });
+
+        try {
+            return remoteClient.getInfo(descriptor);
+        } catch (Exception e) {
+            logger.error("Failed to get remote FlightInfo", e);
```

Comment:
Back to logging, this file:

- `log.error` here
- `log.warn` here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR144 
- `log.debug` here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR117

... for seemingly equal erroneous conditions? may it would make sense to log exception as warn / error?

Reply:
@reta 
 > logger.debug("Failed to parse flight descriptor command", e);

this is more of a 4xx/customer error, so i was thinking of keeping it debug since it can pollute the application log. 

Whereas, below is more an internal error, so I think its important to flag in application log. 
```
            logger.warn("Failed to create proxy producer for remote stream", e);
            throw CallStatus.UNAVAILABLE.withCause(e)
                .withDescription("Unable to create proxy stream: " + e.getMessage())
                .toRuntimeException();
```
Please let me know what do you think?

Reply:
OK, fair, so CallStatus.UNAVAILABLE is an internal error, shouldn't we log it with the same level? 
 - `log.warn` here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR144
 - `log.debug` here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR247

Do we even need to log this errors since I suspect the caller will log them anyway?

Reply:
good point. Let me revisit these, ideally 
 > `log.debug` here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR228 

should be internal error with log.warn

Reply:
> log.warn here https://github.com/opensearch-project/OpenSearch/pull/17446/files#diff-3984143e1a4b52d00d3cddec6def1c2908b60e12ce3cfb254ab5a319926e9d2eR144

This should be internal error as well, since we don't know why it failed. 

I think this should be rule - 

1. Add all errors to `listener`. 
2. All FlightRuntimeException which **are not** `Internal` should not be logged.
3. All FlightRuntimeException which **are** `Internal` should not be logged with `log.error` or `warn` (depending on anticipated severity).

Reply:
Sound very reasonable, thanks @rishabhmaurya

Reply:
done